src/components/layout/Navbar/index.tsx

```diff
@@ -1,18 +1,39 @@
 import { config } from '@/lib';
 import type { PrivateProfile } from '@/lib/types/apiResponses';
 import LightModeLogo from '@/public/assets/acm-logos/general/light-mode.png';
+import ACMIcon from '@/public/assets/icons/acm-icon.svg';
+import CalendarIcon from '@/public/assets/icons/calendar-icon.svg';
+import LeaderboardIcon from '@/public/assets/icons/leaderboard-icon.svg';
 import ProfileIcon from '@/public/assets/icons/profile-icon.svg';
 import ShopIcon from '@/public/assets/icons/shop-icon.svg';
 import Image from 'next/image';
 import Link from 'next/link';
-import { memo } from 'react';
+import { memo, useEffect, useState } from 'react';
+import { useWindowSize } from '../../../lib/utils';
 import DarkModeToggle from '../DarkModeToggle';
 import styles from './style.module.scss';
+import vars from './vars.module.scss';
 
 interface NavbarProps {
   user: PrivateProfile;
 }
 const Navbar = ({ user }: NavbarProps) => {
+  const size = useWindowSize();
+
+  const [menuOpen, setMenuOpen] = useState(false);
+  const [mobile, setMobile] = useState(false);
+  const toggleMenu = () => setMenuOpen(!menuOpen);
+
+  // Switch to mobile less than breakpointMd
+  useEffect(() => {
+    setMobile((size.width || 0) <= parseInt(vars.breakpointMd || '768', 10));
```

Comment:
I mentioned this where you defined the hook but if we can remove the `undefined` from the type signature on `size.width` and initialize it correctly, I think we can avoid needing this default value.

Reply:
Also, be extra careful with assigning a default value this way since `0` is still a falsy value. If you're trying to assign a default value specifically when no value exists (`size.width === undefined`), you should use nullish coalescing (`??`) instead for a stronger condition